lib/_json.py

```diff
@@ -181,18 +181,18 @@ def _get_json_oids(conn_or_curs, name='json'):
     from psycopg2.extras import _solve_conn_curs
 
     conn, curs = _solve_conn_curs(conn_or_curs)
-
-    # Store the transaction status of the connection to revert it after use
-    conn_status = conn.status
-
-    # column typarray not available before PG 8.3
-    typarray = conn.info.server_version >= 80300 and "typarray" or "NULL"
-
-    # get the oid for the hstore
-    curs.execute(
-        "SELECT t.oid, %s FROM pg_type t WHERE t.typname = %%s;"
-        % typarray, (name,))
-    r = curs.fetchone()
+    with curs:
```

Comment:
We don't necessarily want to close the cursor here. We expect the user passed us a cursor they want to keep playing with. Maybe the change should be made in `_solve_conn_curs()`

Reply:
I agree. I've opened #1040 to handle this. Let me know if this follows what you had in mind. If so, I'll rebase this PR.

Reply:
I am sorry, but here I was plain wrong. The cursor is one we are sure we have ownership here, whatever the input of `_solve_conn_curs()`.

As such, I think the changes is #1040 are not needed, and I wouldn't change the semantics of that function - maybe I'd just clarify it. I think the code is simpler using a `with curs` block here, where the cursor is used, rather than make `_solve_conn_curs()` return a context manager.